src/systems/logical_audio_sensor_plugin/Source.hh

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef IGNITION_GAZEBO_SYSTEMS_LOGICAL_AUDIO_SENSOR_PLUGIN_SOURCE_HH_
+#define IGNITION_GAZEBO_SYSTEMS_LOGICAL_AUDIO_SENSOR_PLUGIN_SOURCE_HH_
+
+#include <string>
+#include <unordered_map>
+
+#include <ignition/gazebo/config.hh>
+#include <ignition/math/Vector3.hh>
+
+namespace ignition
+{
+namespace gazebo
+{
+// Inline bracket to help doxygen filtering.
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+namespace systems
+{
+namespace logical_audio
+{
+  /// \brief Audio source attenuation functions.
+  /// AttenuationFunction::Undefined is used to indicate that an
+  /// attenuation function has not been defined yet.
+  enum class AttenuationFunction { Linear, Undefined };
```

Comment:
I believe we have been preferring `ALL_CAPS` for constants in enum classes.

Reply:
I went ahead and changed it to all caps - does this look right now?

https://github.com/ignitionrobotics/ign-gazebo/blob/c664114c3b54ad5c4fd5a9cc8c28c61cfec24089/include/ignition/gazebo/components/LogicalAudio.hh#L38-L46